### ISSUE TITLE
Can we override the build directory when using cmake_external

### ISSUE THREAD

**Issue by deryck1031** [bazelbuild/rules_foreign_cc#334]
the bash script created for building cmake projects using cmake_external always builds in /tmp directory on linux.  Is there a way to override it?

export BUILD_TMPDIR=$(mktemp -d)
export EXT_BUILD_DEPS=$(mktemp -d)

**Comment by irengrig** [comment]
I need to modify the rule to support it (change the shell command to create the temp directory in the specified location, introduce the new rule attribute).
Why do you need it?

**Comment by vrusinov** [comment]
> Why do you need it?

Not OP, but some builds seem to fail when BUILD_TMPDIR and bazel's cache are on different filesystems: https://github.com/bazelbuild/rules_foreign_cc/issues/361

**Comment by vrusinov** [comment]
BTW, workaround of doing an `export TMPDIR=~/tmp` before starting build appears to work

**Comment by wrowe** [comment]
It appears that an --action_env=TMPDIR may be required to pass the TMPDIR value through bazel? After we added this we were able to affect Windows builds by forcing TMPDIR to a better path.

The request is very worthwhile, the current mechanics cause a lot of cruft to accumulate underneath /tmp/ which is never cleaned up. It might be good to consider defaulting to somewhere beneath --output_base so that somewhere along the way, a bazel clean --expunge will rid the machine of these remnants.

It's also critical on windows; this assumption of /tmp lands all bash-behaviors underneath the machine's provisioned msys2 package directory due to cygwin aliasing. When these paths are seen by windows tools, they are interpreted as \tmp under the current drive, which is an entirely different path. This breaks dependencies between foreign_cc packages. It also means that msys2\tmp directory ends up collecting a bunch of cruft (that issue is on msys2/cygwin deployers.) We've found that symlinking msys\tmp -> c:\tmp tends to resolve all of these issues, but that can be hard to accomplish on machines outside of our immediate control.

**Comment by wrowe** [comment]
Note that we were unable to use any specific path without even more pain.

Default behavior
mktemp -> /tmp/tmp.xzy path
bash -> windows both resolve to \tmp of the drive.
Symlink of bash /tmp -> c:\tmp physical directory solves this problem.

Override TMPDIR=c:\path\foreign (to-bazel-build tree)
bash /c/path/foreign -> c:\c\path\foreign is not resolvable.
Symlink of c:\c -> c:\ physical root solves this problem.
Recursive symlinks like this break most applications that would scan the volume.

**Comment by jsharpe** [comment]
The rules no longer use `mktemp` and so this request is no longer needed.
